cpp/cmake/BuildBoost.cmake

```diff
@@ -14,7 +14,7 @@ endif()
 # boost
 add_library(boost INTERFACE)
 add_library(Boost::boost ALIAS boost)
-target_include_directories(boost SYSTEM INTERFACE
+target_include_directories(boost INTERFACE

```

Comment:
did you mean to commit this? because it was added on purpose because the boost headers cause compiler warnings. maybe we need to handle these warnings a different way

Reply:
Changed back. It doesn't compile like this on the cluster. Should I mention that in the document or not?

Reply:
There is an issue related to this: https://github.com/SciCompMod/memilio/issues/853
I haven't checked recently, but it is mentioned there that the build on the cluster works if warnings (or warnings as errors) is disabled. Is that not so? Please add to the other issue if you have seen problems not mentioned there.

I don't think this is related to scorep/profiling, so let's keep it as it is in this merge request.

Reply:
I added a comment there.